docs/www/coprocessors.md

```diff
@@ -0,0 +1,122 @@
+
+```
+redpanda:
+  ...
+  developer_mode: true
+  enable_coproc: true
+  ...
+```
+
+### Creating your first coprocessor
+
+To start, run `rpk wasm generate` in a fresh directory. This will write an example project scaffolding that you can use to get started even quicker.
+
+```
+.
+└── wasm
+    ├── package.json
+    ├── src
+    │   └── wasm.js
+    ├── test
+    │   └── wasm.test.js
+    └── webpack.js
+```
+
+Your source code will be within the `src` directory. Below is an explaination of the API but for now, there is a default operator in there that will consume from one topic called `produce` and produce this onto another topic called `result` while applying a transformation that just uppercases all characters detected on stream.
+
+Once your done modifying the source, cd into the `wasm` directory and run `npm install`. You'll need to do this when you modify your package dependencies. Then run `npm run build` to package up the finished product. You'll see this in the `dist/` folder, with the same name as the source file:
+
+```
+wasm/dist
+└── wasm.js
+```
+
+### Building & launching the WASM engine
+
+To build the wasm engine:
+```
+$ vtools install js-deps && vtools build js
+```
+
+If your using the oss build you can run the `build.sh` script in `$root/src/js`. The products will be placed in `$root/<vbuild or build>/node/output`. 
+
+To start the server, cd into the `output` directory and run:
+```
+$ npm run start <path to redpanda.yaml>
+```
+
+Here's a list of currently supported options that the wasm engine will look for within your `redpanda.yaml` config file:
+
+```
+redpanda:
+  coproc_supervisor_server: 43189
+  ...
+  
+coproc_engine:
+  path: "/home/workspace/coprocessors"
+
+...
+```
+
+You should see three directories be created in your `coproc_engine::path`, `active` `submit` and `inactive`. If you didn't explicitly specify a path in your `redpanda.yaml` file, these folders will exist in their default location: `/var/lib/redpanda/coprocessor`
+
+### Deploying the coprocessor
+
+Currently the coprocessor implementation does not support registering coprocessors that desire to transform a topic that does not yet exist. So you'll have to first create topics for all of the topics you wish to transform. For working with our example that is just the topic named `produce`. For the following examples `rpk` is shown when creating, publishing and consuming from topics, but you can use any kafka api compatible client.
+
+```
+$ rpk api topic create -r 1 produce`
+```
+
+Finally your ready to deploy your coprocessor. Copy the coprocessor from the `wasm/dist/` folder in your project, into the `submit` directory within the wasm engines working path (let's call this the `$wasm_root`). You may need to use sudo depending on where the `$wasm_root` is.
+
+```
+$ cp wasm/dist/wasm.js $wasm_root/submit
```

Comment:
Ok, I thought you didn't need to copy files to each node. This will make it very challenging to run with `rpk container`.

Reply:
As of now you will. Until the new feature gets merged which we're hoping to get in quick